enterprise/metrics/src/main/java/org/neo4j/metrics/source/Neo4jMetricsBuilder.java

```diff
@@ -93,57 +96,61 @@ public boolean build()
         boolean result = false;
         if ( config.get( MetricsSettings.neoTxEnabled ) )
         {
-            life.add( new TransactionMetrics( registry, deps.dataSourceManager(), deps.transactionCounters() ) );
+            life.add( new TransactionMetrics( registry,
+                    dependencies.dataSourceManager(), dependencies.transactionCounters() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoPageCacheEnabled ) )
         {
-
-            life.add( new PageCacheMetrics( registry, deps.pageCacheCounters() ) );
+            life.add( new PageCacheMetrics( registry, dependencies.pageCacheCounters() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoCheckPointingEnabled ) )
         {
             life.add( new CheckPointingMetrics( reporter, registry,
-                    deps.monitors(), deps.checkPointerMonitor() ) );
+                    dependencies.monitors(), dependencies.checkPointerMonitor() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoLogRotationEnabled ) )
         {
             life.add( new LogRotationMetrics( reporter, registry,
-                    deps.monitors(), deps.logRotationMonitor() ) );
+                    dependencies.monitors(), dependencies.logRotationMonitor() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoCountsEnabled ) )
         {
-            life.add( new EntityCountMetrics( registry, deps.idGeneratorFactory() ) );
+            life.add( new EntityCountMetrics( registry, dependencies.entityCountStats() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoNetworkEnabled ) )
         {
-            life.add( new NetworkMetrics( registry, deps.monitors() ) );
+            life.add( new NetworkMetrics( registry, dependencies.monitors() ) );
             result = true;
         }
 
         if ( config.get( MetricsSettings.neoClusterEnabled ) )
         {
-            final ClusterMembers clusterMembers =
-                    resolveClusterMembersOrNull( deps.getDependencyResolver(), logService );
-            if ( clusterMembers != null )
+            boolean isHA = kernelContext.editionName().contains( "HA" );
```

Comment:
Maybe edition should be en enum with `supportsClusterMetrics` property instead of a string?